cfme/fixtures/v2v_fixtures.py

```diff
@@ -0,0 +1,449 @@
+                ]
+                host.update_credentials_rest(credentials=host_data["credentials"])
+
+            rhv_hosts = rhv_provider.hosts.all()
+            for host in rhv_hosts:
+                host_data, = [
+                    data for data in rhv_provider.data["hosts"] if data["name"] == host.name
+                ]
+                host.update_credentials_rest(credentials=host_data["credentials"])
+        except ValueError as error:
+            pytest.skip("Host data invalid for provider: {}".format(error.msg))
+    except Exception:
+        # if above throws ValueError or TypeError or other exception, just skip the test
+        logger.exception("Exception when trying to add the host credentials.")
+        pytest.skip("No data for hosts in providers, failed to retrieve hosts and add creds.")
+    # only yield RHV hosts as they will be required to tag with conversion_tags fixture
+    yield rhv_hosts
+    for host in itertools.chain(rhv_hosts, vmware_hosts):
+        host.remove_credentials_rest()
+
+
+def _tag_cleanup(host_obj, tag1, tag2):
+    """
+        Clean Up Tags
+
+        Returns: Boolean True if all Tags were removed/cleaned
+        or False means all required Tags are present on host.
+    """
+
+    def extract_tag(tag):
+        # Following strip will remove extra asterisk from tag assignment
+        return tag.category.display_name.strip(" *"), tag.display_name
+
+    valid_tags = {extract_tag(tag1), extract_tag(tag2)}
+    tags = host_obj.get_tags()
+    tags_set = set(map(extract_tag, tags))
+    # we always neeed 2 tags for migration, if total is less than 2
+    # don't bother checking what tag was it, just remove it and
+    # then add all required tags via add_tags() call. or if tags on host
+    # are not subset of valid tags, we still remove them.
+    if len(tags_set) < 2 or not tags_set.issubset(valid_tags):
+        host_obj.remove_tags(tags=tags)
+        return True
+    return False
+
+
+@pytest.fixture(scope="module")
+def assign_conversion_tags(request, appliance, host_creds):
+    """Assigning tags to conversation host"""
+    tag1 = appliance.collections.categories.instantiate(
+        display_name="V2V - Transformation Host *"
+    ).collections.tags.instantiate(display_name="t")
+    if hasattr(request, "param") and request.param == "SSH":
+        transformation_method = "SSH"
+    else:
+        transformation_method = "VDDK"
+    tag2 = appliance.collections.categories.instantiate(
+        display_name="V2V - Transformation Method"
+    ).collections.tags.instantiate(display_name=transformation_method)
+    appliance.ssh_client.run_rails_command("'ConversionHost.delete_all'")
+    for host in host_creds:
+        # if _tag_cleanup() returns True, means all tags were removed
+        if _tag_cleanup(host, tag1, tag2):
+            # so we call add_tags to add only required tags
+            host.add_tags(tags=(tag1, tag2))
```

Comment:
Is this adding the categories too? Or do you expect the categories above to already exist?

Reply:
Yes categories are present. V2V - Transformation Host and V2V - Transformation Method